xbmc/addons/interfaces/General.cpp

```diff
@@ -0,0 +1,202 @@
+
+using namespace kodi; // addon-dev-kit namespace
+
+namespace ADDON
+{
+
+void Interface_General::Init(AddonGlobalInterface* addonInterface)
+{
+  addonInterface->toKodi->kodi = static_cast<AddonToKodiFuncTable_kodi*>(malloc(sizeof(AddonToKodiFuncTable_kodi)));
+
+  addonInterface->toKodi->kodi->open_settings_dialog = open_settings_dialog;
+  addonInterface->toKodi->kodi->get_localized_string = get_localized_string;
+  addonInterface->toKodi->kodi->unknown_to_utf8 = unknown_to_utf8;
+  addonInterface->toKodi->kodi->get_dvd_menu_language = get_dvd_menu_language;
+  addonInterface->toKodi->kodi->queue_notification = queue_notification;
+  addonInterface->toKodi->kodi->queue_notification_from_type = queue_notification_from_type;
+  addonInterface->toKodi->kodi->queue_notification_with_image = queue_notification_with_image;
+}
+
+void Interface_General::DeInit(AddonGlobalInterface* addonInterface)
+{
+  if (addonInterface->toKodi && /* <-- needed as long as the old addon way is used */
+      addonInterface->toKodi->kodi)
+  {
+    free(addonInterface->toKodi->kodi);
+    addonInterface->toKodi->kodi = nullptr;
+  }
+}
+
+bool Interface_General::open_settings_dialog(void* kodiBase)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_General::%s - invalid data (addon='%p')", __FUNCTION__, addon);
+    return false;
+  }
+
+  // show settings dialog
+  AddonPtr addonInfo;
+  if (CAddonMgr::GetInstance().GetAddon(addon->ID(), addonInfo))
+  {
+    CLog::Log(LOGERROR, "Interface_General::%s - Could not get addon information for '%s'", __FUNCTION__, addon->ID().c_str());
+  }
+
+  return CGUIDialogAddonSettings::ShowAndGetInput(addonInfo);
+}
+
+char* Interface_General::get_localized_string(void* kodiBase, long dwCode)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_General::%s - invalid data (addon='%p')", __FUNCTION__, addon);
+    return nullptr;
+  }
+
+  if (g_application.m_bStop)
+    return nullptr;
+
+  std::string string;
+  if ((dwCode >= 30000 && dwCode <= 30999) || (dwCode >= 32000 && dwCode <= 32999))
+    string = g_localizeStrings.GetAddonString(addon->ID(), dwCode).c_str();
+  else
+    string = g_localizeStrings.Get(dwCode).c_str();
+
+  char* buffer = strdup(string.c_str());
+  return buffer;
+}
+
+char* Interface_General::unknown_to_utf8(void* kodiBase, const char* source, bool* ret, bool failOnBadChar)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon || !source || !ret)
+  {
+    CLog::Log(LOGERROR, "Interface_General::%s - invalid data (addon='%p', source='%p', ret='%p')", __FUNCTION__, addon, source, ret);
+    return nullptr;
+  }
+   
+  std::string string;
+  *ret = g_charsetConverter.unknownToUTF8(source, string, failOnBadChar);
+  char* buffer = strdup(string.c_str());
+  return buffer;
+}
+
+char* Interface_General::get_dvd_menu_language(void* kodiBase)
```

Comment:
You said that this function is "misused" by addons to obtain Kodi UI language and this is the reason for putting this function to global.general. I suggest to rename (and reimplement) this function to reflect what the addon devs actually need => get_ui_language maybe? 

get_dvd_menu_language should be put into those addon types where dvd menu language actually makes  sense.